include/Accessor/Accessor.h

```diff
@@ -0,0 +1,104 @@
+//=== Accessor.h - Accessor class definition ----*- C++ -*-===//
+//
+// Part of the OMP-UL Project, under the MIT License.
+// See libompx/LICENSE.txt for license information
+//
+//===-------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the Accessor class, which is
+/// used to wrap dynamically allocated regions pointers to provide 
+/// transparent data movement through OpenMP declare mapper
+///
+//===-------------------------------------------------------------===//
+
+
+#ifndef LIBOMPX_ACCESSOR_H
+#define LIBOMPX_ACCESSOR_H
+
+namespace libompx {
+    
+    /// enum to hold different access modes
+    enum access_mode{READ, WRITE, READ_WRITE, ALLOC};
```

Comment:
Maybe rename ALLOC to "TEMPORARY".
Also add documentation for all of them, e.g.
```
READ,   ///< Some short text.
...
```